### PR TITLE
London - Class 7 - Tanika - Javascript Core 1 - Week 2

### PR DESCRIPTION
**Pull request by tanikaDev**
I have completed all the exercises. 

On the extra exercise there was an issue getting the final function working. It states that the helper function is called 'isRadioFrequency' but there wasn't a function with that name. The correct function was 'getAvailableStations' i'm not sure if this was part of the challenge, but I changed the task instructions to state the correct function name hope that was alright.

Details

Name: Tanika
City: London
Slack Name: Tanika Robinson

# Homework Details

Module: Javascript Core 1
Week: 2

### CODE REVIEW

File: exercises/C-comparison-operators/exercise.js [hunk 1]
```diff
@@ -7,14 +7,14 @@
 
 var studentCount = 16;
 var mentorCount = 9;
-var moreStudentsThanMentors; // finish this statement
+var moreStudentsThanMentors = true; // finish this statement
```
Comment (iTrip23) [inline exercises/C-comparison-operators/exercise.js:10]: Hi Tanika,
I have done the exact same thing at first, hardcoded the answers.
```suggestion
var moreStudentsThanMentors = studentCount > mentorCount;
```
You can try and do the same for the rest of the file.
Thanks

File: exercises/D-predicates/exercise.js [hunk 2]
```diff
@@ -2,14 +2,22 @@
...
 // Finish the predicate function to test if the passed number is negative (less than zero)
-function isNegative(number) {}
+function isNegative(number) {
+  return typeof value === "number";
+}
 
 // Finish the predicate function to test if the passed number is between 0 and 10
-function isBetweenZeroAnd10(number) {}
+function isBetweenZeroAnd10(number) {
+  if (number > 0 && number < 10){
```
Comment (iTrip23) [inline exercises/D-predicates/exercise.js:15]: Something to keep in mind is that this statement: `number > 0 && number < 10` is going to return a boolean value(true or false), and you can shorten your code by writing something like:
```suggestion
  function isBetweenZeroAnd10(number) {
  return number > 0 && number < 10;
  }
```
This is just something to keep in mind when writing code. 
Hope this helps.

File: exercises/F-logical-operators/exercise2.js [hunk 3]
```diff
@@ -5,7 +5,37 @@
   Update the code so that you get the expected result.
 */
 
-function isNegative() {}
+function isNegative(num) {
+  if (num < 0) {
```
Comment (iTrip23) [inline exercises/F-logical-operators/exercise2.js:9]: The same principle applies here with the statements in this folder:
`return num < 0;`

**Review by iTrip23 — CHANGES_REQUESTED** [review]
Hi Tanika,
You have done a great job with the homework, all your code is working, I suggested some minor changes. Hope they help you understand the code better.
Happy coding!

File: mandatory/2-function-creation.js [hunk 4]
```diff
@@ -34,10 +58,20 @@ Write a function that:
 - numbers greater 100 must be replaced with 100
 */
 
-function formatPercentage(arr) {}
+function formatPercentage(arr) {
+  for (let i=0; i<arr.length; i++){
+    if (arr[i] > 100) {
+      arr[i] = 100;
+    };
+    arr[i] = parseFloat(arr[i].toFixed(2)) +"%";
```
Comment (Laura-TM) [inline mandatory/2-function-creation.js:66]: I like the use of parseFloat, this is something I need to learn more about.

**Comment by Laura-TM** [comment]
Left some feedback for you, overall, well organised and use of advanced features.

File: extra/1-radio-stations.js [hunk 5]
```diff
@@ -15,17 +15,32 @@
...
+}
+
 /**
  * Next, let's write a function that gives us only the frequencies that are radio stations.
  * Call this function `getStations`.
  *
  * This function should:
  * - Get the available frequencies from `getAllFrequencies`
- * - There is a helper function called isRadioFrequency that takes an integer as an argument and returns a boolean.
+ * - There is a helper function called  'getAvailableStations' that takes an integer as an argument and returns a boolean.
```
[the remark this answers is not in the document]
Comment (ChrisOwen101) [reply extra/1-radio-stations.js:34]: Thanks for spotting Laura and Tanika! I've now updated this coursework. :)

**Pull request closed without merging**